providerkey/download.go

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+
+package providerkey
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+func (pk *providerKey) downloadFile(ctx context.Context, url string) ([]byte, error) {
+	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create HTTP request (%w)", err)
+	}
+
+	res, err := pk.config.HTTPClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to download %s: %w", url, err)
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("status code different from 200 %s: %w", url, err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("status code different from %s %s: %w", http.StatusOK, url, err)
```

NIT: If you end up checking for a different status code first, it can be easy to accidentally leave this the same.

Reply:
Fixed on https://github.com/opentofu/libregistry/pull/47/commits/0428656cc81ae6b127f44719f6c77e081b55d3f0